temporalio/workflow.py

```diff
@@ -1045,6 +1187,16 @@ def _apply_to_class(
                     )
                 else:
                     queries[query_defn.name] = query_defn
+            elif isinstance(member, UpdateMethodMultiArg):
```

Comment:
Mentioned before, but I wonder if it's simpler to take the named definition approach instead of runtime checkability of a protocol (which is less commonly used in Python)

Reply:
Simpler in what way? To me this reads as more obvious.

Reply:
Simpler from a reader POV in that this is just checking that property and a couple of methods exist on the object (and not checking types of those) instead of it actually being an instance of the type as `isinstance` usually implies. Runtime checkability via `isinstance` comes across as doing the same thing (just checking attributes at runtime) but harder to recognize that's what's happening.

Having said that, since it's all hidden from the user, I don't have a strong opinion here if it works with Python 3.7.

Reply:
Confirmed in another comment to work with 3.7, good enough for me.